java/java-guestbook/backend/Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Use maven to compile the java application.
+FROM docker.io/maven AS build-env
+
+# Set the working directory to /app
+WORKDIR /app
+
+# copy the pom.xml file to download dependencies
+COPY pom.xml ./
+
+# download dependencies as specified in pom.xml
+RUN mvn verify --fail-never
```

Comment:
Why do you do this in two steps?  it seems like you could just do a full copy, then mvn package.

Reply:
Doing it all in one step causes a ~40 second build/deploy time for any small text change. If we do "mvn verify" to pull the dependencies on a lower layer, it only takes ~5 seconds to re-package the jar file when you make small changes to the Java code

That being said, I'm not too familiar with maven. Is there a more concise way to break apart the fetching dependencies and packaging jar steps, rather than using "mvn verify" and "mvn package"?